book-2-mvc/chapters/ASPNET_INTRO.md

```diff
@@ -77,63 +89,136 @@ public SqlConnection Connection
+        using (SqlCommand cmd = conn.CreateCommand())
+        {
+            cmd.CommandText = @"
+                SELECT Id, [Name], ImageUrl, NeighborhoodId
+                FROM Walker
+                WHERE Id = @id
+            ";
+
+            cmd.Parameters.AddWithValue("@id", id);
+
+            SqlDataReader reader = cmd.ExecuteReader();
+
+            if (reader.Read())
+            {
+                Walker walker = new Walker
+                {
+                    Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                    Name = reader.GetString(reader.GetOrdinal("Name")),
+                    ImageUrl = reader.GetString(reader.GetOrdinal("ImageUrl")),
+                    NeighborhoodId = reader.GetInt32(reader.GetOrdinal("NeighborhoodId"))
+                };
+
+                reader.Close();
+                return View(walker);
+            }
+            else
+            {
+                return NotFound();
+            }
+        }
+    }
+}
+```
+
+Notice that this method accepts an `id` parameter. When the ASP<span>.NET</span> framework invokes this method for us, it will take whatever value is in the url and pass it to the `Details` method. For example, if the url is `walkers/details/2`, the framework will invoke the Details method and pass in the value `2`. The code looks in the database for a walker with the id of 2. If it finds one, it will return it to the view. If it doesn't the user will be given a 404 Not Found page.
 
-## Viewing the List of Students
+Right click the Details method and select Add View. Keep the name "Details", select "Details" for the Template dropdown, and select "Walker" for the model class. Make the same changes in the view as before and replace the image url with the image tag
 
-Now you start the project. Visit [http://localhost:5000/students](http://localhost:5000/students) in your browser to see the HTML representation of students you have in your database.
+```html
+<img class="bg-info" src="@Model.ImageUrl" alt="avatar" />
+```
 
-## Adding the Other Views
+Run the application and go to `/walkers/details/1`. Then go to `/walkers/details/999` to see that we get a 404 response back.
 
-The next step is to create the `Details` view, the `Edit` view, and the `Delete` view. Use the scaffolding for _MVC View_ to create the Razor templates for each one. Don't attempt to build the create view until you get to the next chapter about View Models.
 
+## Exercise
 
+Create an Owners controller and implement the `Index` and `Details` methods. When viewing the details page of an owner, list all the dogs for that owner. Afterwards, go into the `Shared` folder in the `_Layout.cshtml` file. Add links for "Walkers" and "Owners" in the navbar. If you finish, try changing the views and the styling to your liking. By default, Visual Studio installs bootstrap and jQuery to the project. You're welcome to embrace this if you like, or use something completely different--it's entirely up to you.
```

Comment:
> You're welcome to embrace this if you like, or use something completely different--it's entirely up to you.

I think it's too soon to give them this freedom. It's unnecessary cognitive load at this point.

Reply:
I think most students will choose to ignore it but I want to let them know early on that this is not how MVC application _have_ to look

Reply:
I think it's a good idea to mention it in the chapter, but putting in the exercise might make it seem like we are recommending that students make styling changes. Any student who is ready to hear that MVC can be styled will (hopefully) get that from the chapter/lecture.